### ISSUE TITLE
ipmitool 1.8.18 segmentation fault due to incorrect compiler flags

### ISSUE THREAD

**Issue by craigafinch** [Homebrew/homebrew-core#9423]
# Please follow the general troubleshooting steps first:

- [ x  ] Ran `brew update` and retried your prior step?
- [ x ] Ran `brew doctor`, fixed as many issues as possible and retried your prior step?
- [ x ] Confirmed this is problem with specific formulae and not Homebrew/brew? If it's a general Homebrew/brew problem please file this issue at https://github.com/Homebrew/brew/issues/new

### Bug reports:

ipmitool was installed successfully, but segfaults after prompting for a password. The command line that causes the issue is:

```
ipmitool -I lanplus -H <ip address> -U <username> sol activate
Password: 
Segmentation fault: 11
```

This is a known problem, which can be resolved by compiling with ```GNU99``` compiler flag instead of ```C99```:
https://sourceforge.net/p/ipmitool/bugs/436/

```
HOMEBREW_VERSION: 1.1.8
ORIGIN: https://github.com/Homebrew/brew
HEAD: e59ada508727f11464595893783beb914c26f60b
Last commit: 6 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: efdf63c381263be40b8ae8bc3fe30c865f9f39de
Core tap last commit: 67 minutes ago
HOMEBREW_PREFIX: /usr/local
HOMEBREW_REPOSITORY: /usr/local/Homebrew
HOMEBREW_CELLAR: /usr/local/Cellar
HOMEBREW_BOTTLE_DOMAIN: https://homebrew.bintray.com
CPU: quad-core 64-bit haswell
Homebrew Ruby: 2.0.0-p648
Clang: 8.0 build 800
Git: 2.11.0 => /usr/local/bin/git
Perl: /usr/bin/perl
Python: /usr/bin/python
Ruby: /usr/bin/ruby => /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby
Java: N/A
macOS: 10.12.3-x86_64
Xcode: 8.2.1
CLT: N/A
X11: N/A
```

```
$ brew doctor
Your system is ready to brew.
```

```
$ brew gist-logs ipmitool
Error: No logs.
```

**Comment by MikeMcQuaid** [comment]
> This is a known problem, which can be resolved by compiling with GNU99 compiler flag instead of C99:
> https://sourceforge.net/p/ipmitool/bugs/436/

Could you try and open a pull request? [This document](http://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request.html) should help and we're happy to walk you through anything else.

Thanks!

**Comment by adaugherity** [comment]
Unfortunately ipmitool is still broken.  The fix did nothing -- the compiler option should be `-std=gnu99` (one hyphen, not two), and it was redundant, since ipmitool version 1.8.18 is already defaulting to `gnu99` (1.8.16 was the only version using `c99`).

I think the upstream issue may have been prematurely closed, because while they did end up changing to gnu99, it's still broken.  I finally discovered that the segfault is due to missing a function prototype for getpass(), so apparently the pointer it returns is being mishandled.  

Debian fixed it [a different way](https://anonscm.debian.org/cgit/collab-maint/ipmitool.git/tree/debian/patches/0110-getpass-prototype.patch) (and that patch is still used in their package).  I tried that fix, but it did not fix it, as it is apparently only applicable to glibc, and not the macOS SDK.

I eventually discovered that the function prototype for getpass() (in `$(xcrun --show-sdk-path)/usr/include/unistd.h`) is wrapped in a block where it is only used 
`#if !defined(_POSIX_C_SOURCE) || _POSIX_C_SOURCE < 200112L`. 

`_POSIX_C_SOURCE` is in turned defined in `<sys/cdefs.h>` according to the value of `_XOPEN_SOURCE`.  ipmitool has `#define _XOPEN_SOURCE 700` at the top of `lib/ipmi_main.c`.  Changing this to 500 (which in turn means _POSIX_C_SOURCE=199506L, and getpass() included) makes the segfault go away.

I don't know if that's the proper fix, but it works for me.
